packages/select/src/components/query-list/queryList.tsx

```diff
@@ -419,10 +428,12 @@ export class QueryList<T> extends AbstractComponent2<QueryListProps<T>, IQueryLi
         if (this.itemsParentRef != null) {
             if (isCreateNewItem(activeItem)) {
                 const index = this.isCreateItemFirst() ? 0 : this.state.filteredItems.length;
-                return this.itemsParentRef.children.item(index) as HTMLElement;
+                return this.itemRefs.get(index) ?? (this.itemsParentRef.children.item(index) as HTMLElement);
```

Comment:
The fallback is for backwards compat (if a user didn't wire up the new `ref` to `elementRef` on the `MenuItem` for example)